text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,109 @@
+# Semantic Versioning of Buildpacks and Builders
+
+## Summary
+
+All Paketo buildpack releases use the semver syntax for versioning, but does not necessarily follow semver semantics.
+
+The [CNB Buildpacks spec](https://github.com/buildpacks/spec/blob/buildpack/v0.6/buildpack.md) says:
+
+> The buildpack version:
+>
+> - MUST be in the form <X>.<Y>.<Z> where X, Y, and Z are non-negative integers and must not contain leading zeros.
+>    - Each element MUST increase numerically.
+>    - Buildpack authors will define what changes will increment X, Y, and Z.
+
+
+It does not define what changes will increment the Major (`X`), Minor (`Y`) and Patch (`Z`) versions
+of buildpack releases.
+
+This RFC attempts to set guidelines and good practices for Paketo Buildpack authors/maintainers to follow
+when a buildpack release has to be created.
+
+## Motivation
+
+As a Buildpack/Builder maintainer,<br/>
+I need a release versioning guideline,
+so I can make my release versions consistent.
+
+As a Buildpack/Builder user,<br/>
+I want to know how versioning is done,
+so I can use the appropriate version and
+decide when to upgrade.
+
+## Semantics
+
+Semantic Versioning 2.0.0 defines versioning syntax as follows [semver.org](https://semver.org/):
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+> 
+>   - MAJOR version when you make incompatible API changes,
+>   - MINOR version when you add functionality in a backwards compatible manner, and
+>   - PATCH version when you make backwards compatible bug fixes.
+
+The proposal is to conform to the above semantics as much as possible.
+
+### Changes
+
+In terms of Paketo Buildpacks/Builders, this translates to:
+
+* Bumping MAJOR is analogous to an "`app or related buildpacks may not work without change`" relation between releases.
+* Bumping MINOR is analogous to an "`updates or adds features`" relation between releases.
+* Bumping PATCH is analogous to a "`fixes issue`" relation between releases.
+
+The following table lists the usual circumstances/events where bumping a particular component is expected:
+
+#### Buildpack
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| \<Any incompatible change\> | \<Feature additions but compatible with previous release\> | \<Fixes but no major feature changes\> |
+| Change in Buildpack ID | Addition/removal of dependency<br>(when not superseded by conditions in the Major column) | Bug fixes |
+| Change in supported Stacks | Buildpack runs a new command the causes backward compatible<br>change in behavior (e.g. x-install buildpack adds running "x clean") | Performance improvement, internal tweaks |
```

Comment:
>Change in supported Stacks

I understand why the removal of a stack would trigger a major bump, but why would the addition of a stack? That seems like it could be under a minor version since it's just adding functionality and not likely to break something.

My thought would be to change this to "Removal of a stack or renaming of an existing stack". Then specifically add "Adding a stack to the "Minor" column, to indicate additions can be done under a minor bump.

Reply:
Fixed